site/cds_rdm/views.py

```diff
@@ -8,19 +8,46 @@
 
 """Additional views."""
 
-from flask import Blueprint, redirect
+from flask import Blueprint, current_app, redirect, render_template, request, url_for
+from invenio_rdm_records.proxies import current_rdm_records
+from invenio_rdm_records.records.api import RDMRecord
 from invenio_rdm_records.resources.urls import record_url_for
+from sqlalchemy.orm.exc import NoResultFound
 
-from cds_rdm.resolver import get_pid_by_legacy_recid
+from cds_rdm.resolver import get_pid_by_legacy_recid, get_record_by_version
 
 
-def legacy_redirect(id):
+def not_found_error(error):
+    """Handler for 'Not Found' errors."""
+    return render_template(current_app.config["THEME_404_TEMPLATE"]), 404
+
+
+def legacy_redirect(legacy_id):
     """Redirect legacy recid."""
-    pid = get_pid_by_legacy_recid(id)
+    pid = get_pid_by_legacy_recid(legacy_id)
     url_path = record_url_for(pid_value=pid)
     return redirect(url_path)
 
 
+def legacy_files_redirect(legacy_id, filename):
+    """Redirection for legacy files."""
+    parent_pid = get_pid_by_legacy_recid(legacy_id)
+    query_params = request.args.copy()
```

Comment:
nit: it would be a bit cleaner to wrap this in a decorator (check invenio-app-rdm -> records_ui->views->decorators.py)

Reply:
Why a decorator? This is used only in one specific view.